src/components/calltree/ProfileCallTreeContextMenu.js

```diff
@@ -167,14 +163,58 @@ class ProfileCallTreeContextMenu extends PureComponent {
       case 'merge-function':
         addTransformToStack(threadIndex, {
           type: 'merge-function',
-          funcIndex: selectedCallNodePath[selectedCallNodePath.length - 1],
+          funcIndex: selectedFunc,
         });
         break;
+      case 'collapse-resource': {
+        const { funcTable } = thread;
+        const resourceIndex = funcTable.resource[selectedFunc];
+        // A new collapsed func will be inserted into the table at the end. Deduce
+        // the index here.
+        const collapsedFuncIndex = funcTable.length;
+        addTransformToStack(threadIndex, {
+          type: 'collapse-resource',
+          resourceIndex,
+          collapsedFuncIndex,
+          implementation,
+        });
+        break;
+      }
       default:
         throw new Error('Type not found.');
     }
   }
 
+  getNameForResouce(): string | null {
+    const {
+      selectedCallNodePath,
+      thread: { funcTable, stringTable, resourceTable, libs },
+    } = this.props;
+
+    const funcIndex = selectedCallNodePath[selectedCallNodePath.length - 1];
+    if (funcIndex === undefined) {
+      return null;
+    }
+    const isJS = funcTable.isJS[funcIndex];
+
+    if (isJS) {
+      const fileNameIndex = funcTable.fileName[funcIndex];
+      return fileNameIndex === null
+        ? null
+        : stringTable.getString(fileNameIndex);
+    } else {
```

Comment:
No else after return please.

Reply:
should we enable this eslint rule ? https://eslint.org/docs/rules/no-else-return